### ISSUE TITLE
ios analyzeImage method  Cannot recognize this image QR code

### ISSUE THREAD

**Issue by assemblyios** [juliansteenbakker/mobile_scanner#849]
ios analyzeImage method  Cannot recognize this image QR code

![企业微信20231106-180927](https://github.com/juliansteenbakker/mobile_scanner/assets/12081932/21360c68-8242-48b8-ae94-c7db6b30474a)

**Comment by assemblyios** [comment]
But a scan can identify it

**Comment by navaronbracke** [comment]
So `analyzeImage()` returns false, but `controller.barcodes.listen()` does provide the barcode?

**Comment by assemblyios** [comment]
> So `analyzeImage()` returns false, but `controller.barcodes.listen()` does provide the barcode?

controller.barcodes.listen()  Do not provide any  barcode

**Comment by navaronbracke** [comment]
If `analyzeImage()` returns false and `controller.barcodes` does not include it, this is working as intended.

As I said before, I hope to fix the signature of this API, because it is a bit confusing at the moment.

I'll try to test scanning with the given barcode, maybe it has to do with specific options for the scanner.

**Comment by assemblyios** [comment]
> If `analyzeImage()` returns false and `controller.barcodes` does not include it, this is working as intended.
> 
> As I said before, I hope to fix the signature of this API, because it is a bit confusing at the moment.
> 
> I'll try to test scanning with the given barcode, maybe it has to do with specific options for the scanner.

Thank you for your reply and have a nice day

**Comment by jgaviriadev** [comment]
you have to convert the generated image, is the solution I found. The error occurs when the QR is generated from the same cell phone on iOS. It is an error specific to the package that generates the qr or ios. So, the solution is that, as soon as you have the path, you convert the image and save it in a temporary path and pass this new path to analyzeImage()

**Comment by dahvlhldle** [comment]
> you have to convert the generated image, is the solution I found. The error occurs when the QR is generated from the same cell phone on iOS. It is an error specific to the package that generates the qr or ios. So, the solution is that, as soon as you have the path, you convert the image and save it in a temporary path and pass this new path to analyzeImage()

You are correct. How to create a new file? I use the file copy and save the copied file to the cache, but it does not work. I change the crop a image, and then use the new file to analyze the image, and it succeeds.

**Comment by Blue-Cheesecake** [comment]
> > you have to convert the generated image, is the solution I found. The error occurs when the QR is generated from the same cell phone on iOS. It is an error specific to the package that generates the qr or ios. So, the solution is that, as soon as you have the path, you convert the image and save it in a temporary path and pass this new path to analyzeImage()
> 
> You are correct. How to create a new file? I use the file copy and save the copied file to the cache, but it does not work. I change the crop a image, and then use the new file to analyze the image, and it succeeds.

Could you provide how you crop an image? I tried it but it didn't work.

**Comment by juliansteenbakker** [comment]
I'm closing this as there is no activity, and newer versions of this plug are already released. If this still happens with the latest version let me know by commenting.
